src/screens/activity/ActivityRow.tsx

```diff
@@ -80,53 +80,60 @@ export const ActivityBasicRow = ({
   // USD Balance
   const usdBalance = roundBalance(price, 2)
 
-  const txSummary: TransactionSummaryScreenProps = useMemo(
-    () => ({
+  const txSummary: TransactionSummaryScreenProps = useMemo(() => {
+    const tokenUsd = usdBalance.toFixed(2)
+    const feeUsd = Number(fee.usdValue).toFixed(2)
+    const totalUsd = (Number(tokenUsd) + Number(feeUsd)).toFixed(2)
+    const isAmountSmall = !usdBalance && !!price
```

Comment:
This code is repeated a few times in this PR. However, this type of code shouldn't be at the component level, we should outsource it. 

Take a look at the existing functions for handling numbers in the shared/utils. I think there should be something that handles this,